parquet-hadoop/src/main/java/org/apache/parquet/hadoop/util/ColumnMasker.java

```diff
@@ -0,0 +1,274 @@
+      }
+
+      writer.endBlock();
+      store = reader.readNextRowGroup();
+      blockIndex++;
+    }
+  }
+
+  private void processChunk(ColumnDescriptor descriptor, ColumnChunkMetaData chunk, ColumnReadStoreImpl crStore,
+                            TransParquetFileReader reader, TransParquetFileWriter writer, MessageType schema,
+                            Set<ColumnPath> paths, MaskMode maskMode) throws IOException {
+    reader.setStreamPosition(chunk.getStartingPos());
+
+    if (paths.contains(chunk.getPath())) {
+      if (maskMode.equals(MaskMode.NULLIFY)) {
+        Type.Repetition repetition = descriptor.getPrimitiveType().getRepetition();
+        if (repetition.equals(Type.Repetition.REQUIRED)) {
+          throw new IOException("Required column [" + descriptor.getPrimitiveType().getName() + "] cannot be nullified");
+        }
+        nullifyColumn(descriptor, chunk, crStore, writer, schema);
+      } else {
+        throw new UnsupportedOperationException("Only nullify is supported for now");
+      }
+    } else {
+      BloomFilter bloomFilter = reader.readBloomFilter(chunk);
+      ColumnIndex columnIndex = reader.readColumnIndex(chunk);
+      OffsetIndex offsetIndex = reader.readOffsetIndex(chunk);
+      writer.appendColumnChunk(descriptor, reader.getStream(), chunk, bloomFilter, columnIndex, offsetIndex);
+    }
+  }
+
+  private void nullifyColumn(ColumnDescriptor descriptor, ColumnChunkMetaData chunk, ColumnReadStoreImpl crStore,
+                             TransParquetFileWriter writer, MessageType schema) throws IOException {
+    long totalChunkValues = chunk.getValueCount();
+    int dMax = descriptor.getMaxDefinitionLevel();
+    ColumnReader cReader = crStore.getColumnReader(descriptor);
+
+    writer.startColumn(descriptor, totalChunkValues, CompressionCodecName.UNCOMPRESSED);
+
+    WriterVersion writerVersion = chunk.getEncodingStats().usesV2Pages() ? WriterVersion.PARQUET_2_0 : WriterVersion.PARQUET_1_0;
+    ParquetProperties props = ParquetProperties.builder()
+      .withWriterVersion(writerVersion)
+      .build();
+    ColumnWriter cWriter = props.newColumnWriteStore(schema, new DummyPageWriterStore()).getColumnWriter(descriptor);
+
+    for (int i = 0; i < totalChunkValues; i++) {
+      int rlvl = cReader.getCurrentRepetitionLevel();
+      int dlvl = cReader.getCurrentDefinitionLevel();
+      if (dlvl == dMax) {
+        // since we checked ether optional or repeated, dlvl should be > 0
+        if (dlvl == 0) {
+          throw new IOException("definition level is detected to be 0 for column " + chunk.getPath().toDotString() + " to be nullified");
+        }
+        // we just write one null for the whole list at the top level, instead of nullify the elements in the list one by one
+        if (rlvl == 0) {
+          cWriter.writeNull(rlvl, dlvl - 1);
+        }
+      } else {
+        cWriter.writeNull(rlvl, dlvl);
+      }
+    }
+
+    BytesInput data = cWriter.concatWriters();
```

Comment:
So the concept is to write the nulls into a "hacked" `ColumnWrite`r and extract the bytes from it then write it as a page. This way you write the whole column chunk into one page. This breaks the concept of column-indexes that we write new pages for a couple of rows (default is 20k) even if the pages are small in bytes.
I think, it would be much better to use a proper `PageWriteStore` instead of a dummy one and let the original implementation decide when to create new pages. This way you do not need to extend the existing interface with `contactWriters` (which also breaks the current concept of the interface that is only a sink for the data).

Reply:
The problem is we need to make the class ColumnChunkPageWriteStore public because we are under the namespace of org.apache.parquet.hadoop.util which cannot use ColumnChunkPageWriteStore for now.

Reply:
We already have a lot of classes that are public because of we cannot fine-tune the java visibility but are for internal use only. I would suggest make it public if it is required for your design and comment that it is not for public use. You may want to use the yetus' `@Private` annotation as well.

Reply:
sure